crates/bevy_render/src/render_resource/pipeline_cache.rs

```diff
@@ -85,9 +85,20 @@ impl ShaderCache {
                         return Err(RenderPipelineError::AsModuleDescriptorError(err, processed));
                     }
                 };
-                entry.insert(Arc::new(
-                    render_device.create_shader_module(&module_descriptor),
-                ))
+
+                render_device
+                    .wgpu_device()
+                    .push_error_scope(wgpu::ErrorFilter::Validation);
+                let shader_module = render_device.create_shader_module(&module_descriptor);
+                use futures_util::future::FutureExt;
+                let error = render_device.wgpu_device().pop_error_scope();
+                if let Some(Some(wgpu::Error::Validation { description, .. })) =
+                    error.now_or_never()
```

Comment:
`now_or_never` will return `Some` if the future is ready and `None` otherwise.
On native wgpu will yield the error immediatly while on wasm it may take some time.

I used it here to not make the pipeline cache more complicated by storing and polling futures, which is IMO not worth it for this change.